src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.GenerateAssemblyInfo.targets

```diff
@@ -115,6 +115,15 @@ Copyright (c) .NET Foundation. All rights reserved.
         <_Parameter2>%(AssemblyMetadata.Value)</_Parameter2>
       </AssemblyAttribute>
     </ItemGroup>
+
+    <ItemGroup Condition="'$(TargetPlatformIdentifier)' != ''
+                         and '$(MinimumOSPlatform)' != ''
+                         and '$(TargetFrameworkIdentifier)' == '.NETCoreApp'
+                         and $([MSBuild]::VersionGreaterThanOrEquals($(_TargetFrameworkVersionWithoutV), '5.0'))">
```

Comment:
@dsplaisted @mhutch I added the condition on > net5.0 . We won't have the type when it is lower